### ISSUE TITLE
Incorrect JUnit XML report being generated

### ISSUE THREAD

**Issue by jiyoon-sb** [webdriverio/webdriverio#6014]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.6.2
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 12.13.0
 - **NPM version:** 6.12.0
 - **Browser name and version:** N/A
 - **Platform name and version:** N/A
 - **Additional wdio packages used (if applicable):** @wdio/cli, @wdio/dot-reporter, @wdio/junit-reporter, @wdio/local-runner, @wdio/mocha-framework, @wdio/reporter, @wdio/sauce-service, @wdio/selenium-standalone-service, @wdio/spec-reporter, @wdio/sync

**Config of WebdriverIO**
```
'use strict';
 
const chai = require('chai');
 
global.assert = chai.assert;
global.chai = chai;
chai.Should();
 
exports.config = {
  user: process.env.SAUCE_USERNAME,
  key: process.env.SAUCE_ACCESS_KEY,
  region: 'us-west-1',
 
  maxInstances: 1,
  capabilities: [
      {
        browserName: 'chrome',
        platformName: 'Windows 10',
        browserVersion: '83.0',
        'sauce:options': {
          parentTunnel: '<insert tunnel name>',
          public: 'team',
          screenResolution: '1280x960',
          tunnelIdentifier: 'Primary',
          extendedDebugging: true,
        }
      }
  ],
  execArgv: [],
  sync: true,
  logLevel: 'trace',
  bail: 0,
  screenshotPath: 'screenshots',
  baseUrl: '',
  outputDir: './wdio-log',
  waitforTimeout: 60000,
  connectionRetryTimeout: 120000,
  connectionRetryCount: 3,
  services: [
    [
      'sauce',
      {
        sauceConnect: true,
        sauceConnectOpts: {
          parentTunnel: '<insert tunnel name here>',
          tunnelIdentifier: 'Primary',
        },
      },
    ],
  ],
  framework: 'mocha',
  reporters: [
    [
      'junit',
      {
        outputDir: './reports/',
        outputFileFormat: function (opts) {
          return `results-${opts.cid}.${opts.capabilities.browserName}-v${opts.capabilities.browserVersion}.xml`;
        },
      },
    ],
  ],
 
  mochaOpts: {
    retries: 2,
    timeout: 0,
   ui: 'bdd',
  }
};
```

**Describe the bug**
With webdriverio and accompanying packages @6.5.2 and below, our team noticed that tests that initially failed and passed on a retry, the junit report shows the "failed" tests as being `<skipped/>`.
Starting at packages @6.6.0 and above, we are now seeing the tests that have initially failed and passed on a retry are now being counted as an error/failure.
We then get an incorrect failure count being reported when in fact the test has passed.

**To Reproduce**
1. Set up wdio configuration to allow for retries.
2. Set up a test to fail and then pass on a retry.

```
'use strict';
 
describe(`custom-test`, function () {
  let tries = 0;

  it(`passes`, function () {
    if (tries < 1) {
      ++tries;
      chai.expect(true).to.be.false;
    }
 
    chai.expect(true).to.be.true;
  });
});
```

**Expected behavior**
Only the last run of a test should get an entry in the junit report.  In other words, there should not be multiple entries for the same test, only the final result of the test case matters.

**Log**
- [wdio.log](https://gist.github.com/jiyoon-sb/5ab8882a47dbc4694fefdaed02d1f80b#file-wdio-log)
- [wdio-0-0.log](https://gist.github.com/jiyoon-sb/464981fec3a111fd0cc064d8f96dbd42#file-wdio-0-0-log)

**Additional context**
The combination of these package versions produces a [junit-report.xml](https://gist.github.com/jiyoon-sb/fca52528581c5bdd355e3be14e2b498f#file-junit-report-before-xml) with the `<skipped/>` output:
```
"@wdio/cli": "6.5.2",
"@wdio/dot-reporter": "6.4.7",
"@wdio/firefox-profile-service": "6.4.7",
"@wdio/junit-reporter": "6.5.1",
"@wdio/local-runner": "6.5.2",
"@wdio/mocha-framework": "6.5.0",
"@wdio/reporter": "6.4.7",
"@wdio/sauce-service": "6.5.0",
"@wdio/selenium-standalone-service": "6.4.7",
"@wdio/spec-reporter": "6.4.7",
"@wdio/sync": "6.5.0",
"webdriverio": "6.5.2",
```

The combination of these package versions produces a [junit-report.xml](https://gist.github.com/jiyoon-sb/a24bc0be21a281d7bd1d0630185212d9#file-junit-report-after-xml) with the increased error/failure count:
```
"@wdio/cli": "6.6.0",
"@wdio/dot-reporter": "6.6.0",
"@wdio/firefox-profile-service": "6.6.0",
"@wdio/junit-reporter": "6.6.0",
"@wdio/local-runner": "6.6.0",
"@wdio/mocha-framework": "6.6.0",
"@wdio/reporter": "6.6.0",
"@wdio/sauce-service": "6.6.0",
"@wdio/selenium-standalone-service": "6.6.0",
"@wdio/spec-reporter": "6.6.0",
"@wdio/sync": "6.6.0",
"webdriverio": "6.6.0",
```

I attempted to pinpoint a specific package by upgrade one at a time (junit-reporter for example) but could not repro.  It appears to be a combination of the above set.

**Comment by christian-bromann** [comment]
Hey,
thanks for filing the issue. This seems indeed like bug. We would appreciate any help on this so please help us out! Have a look at our [contributing guidelines](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) and reach out if you have any questions. Cheers!

**Comment by cjatkinson** [comment]
Investigating

**Comment by lthurr** [comment]
I think this could be fixed by https://github.com/webdriverio/webdriverio/issues/7104

**Comment by christian-bromann** [comment]
fixed by #7157
